src/dao/pages/ProposalDetailsPage/useProposalDetails.tsx

```diff
@@ -1,12 +1,14 @@
-import { useDAOs, useDAOProposals } from "@dao/hooks";
+import { useDAOs, useDAOProposals, useContract } from "@dao/hooks";
 import { MultiSigDAODetails } from "@dao/services";
 import { isNotNil } from "ramda";
 
 export function useProposalDetails(daoAccountId: string, transactionHash: string) {
-  const daosQueryResults = useDAOs<MultiSigDAODetails>(daoAccountId);
+  const daoAccountIdQueryResults = useContract(daoAccountId);
+  const daoAccountEVMAddress = daoAccountIdQueryResults.data?.data.evm_address;
+  const daosQueryResults = useDAOs<MultiSigDAODetails>();
   const { data: daos } = daosQueryResults;
-  const dao = daos?.find((dao: MultiSigDAODetails) => dao.accountId === daoAccountId);
-  const daoProposalsQueryResults = useDAOProposals(daoAccountId, dao?.safeId ?? "");
+  const dao = daos?.find((dao) => dao.accountId.toLowerCase() == daoAccountEVMAddress?.toLowerCase());
```

Comment:
Use [Strict equality (===)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality) not [Equality (==)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality). The Mozilla developer doc links explain the difference. There is rarely ever a reason to use `==` for typical value equality checks.

Reply:
Done.